src/WireMock.Net/Matchers/Request/RequestMessageBodyMatcher.cs

```diff
@@ -142,20 +142,20 @@ private double CalculateMatchScore(RequestMessage requestMessage)
             {
                 return Matchers.Max(matcher => CalculateMatchScore(requestMessage, matcher));
             }
-
-            if (Func != null)
+            
+            if (Func != null && requestMessage?.BodyData?.BodyAsString != null)
```

Comment:
With this extra `&&`, the Func is not executed anymore when the BodyAsString is null.
It could be that you actually want act on this. So better to keep original if-statement, and make sure that the function-call is null-safe. See comment below.